apache/yunikorn-core
# [YUNIKORN-2209] Remove limit checks in QueueTracker

### What is this PR for?
If there is no limit configured for user but wild card has some limit, then use that limit for all users permanently till next config change. Since limit has been defined in corresponding user queue tracker object, there is no need to wild card checks during headroom, canrunapp and increase calls etc thereby increases the performance of critical path. While clearing old configs, in case wild card user limit has been removed, ensure all users used that wild card user limit settings has reset to its original value. Introduced "useWildCard" bool flag in queue tracker to handle this.

### What type of PR is it?

### Todos

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2209

### How should this be tested?
Existing unit tests should continue to run as usual without any failures.
Added new unit test to cover this change.

### Screenshots (if appropriate)

### Questions:
